parsec/core/types/manifest.py

```diff
@@ -54,994 +56,52 @@
-            # TODO: Should this assert be in remote workspace manifest definition instead ?
-            # TODO: but in theory remote workspace manifest should assert version > 0 !
-            assert data["base"].version != 0 or not data["base"].workspaces
-            return LocalUserManifest(**data)
-
-    base: RemoteUserManifest
-
-    last_processed_message: int
-    workspaces: Tuple[WorkspaceEntry, ...]
-
-    speculative: bool
-
-    @classmethod
-    def new_placeholder(
-        cls, author: DeviceID, timestamp: DateTime, id: EntryID = None, speculative: bool = False
-    ) -> "LocalUserManifest":
-        return cls(
-            base=RemoteUserManifest(
-                author=author,
-                timestamp=timestamp,
-                id=id or EntryID.new(),
-                version=0,
-                created=timestamp,
-                updated=timestamp,
-                last_processed_message=0,
-                workspaces=(),
-            ),
-            need_sync=True,
-            updated=timestamp,
-            last_processed_message=0,
-            workspaces=(),
-            speculative=speculative,
-        )
-
-    # Helper
-
-    def get_workspace_entry(self, workspace_id: EntryID) -> WorkspaceEntry:
-        return next((w for w in self.workspaces if w.id == workspace_id), None)
-
-    # Evolve methods
-
-    def evolve_workspaces_and_mark_updated(self, timestamp: DateTime, *data) -> "LocalUserManifest":
-        workspaces = {**{w.id: w for w in self.workspaces}, **{w.id: w for w in data}}
-        return self.evolve_and_mark_updated(
-            workspaces=tuple(workspaces.values()), timestamp=timestamp
-        )
-
-    def evolve_workspaces(self, *data) -> "LocalUserManifest":
-        workspaces = {**{w.id: w for w in self.workspaces}, **{w.id: w for w in data}}
-        return self.evolve(workspaces=tuple(workspaces.values()))
-
-    # Remote methods
-
-    @classmethod
-    def from_remote(cls, remote: RemoteUserManifest) -> "LocalUserManifest":
-        return cls(
-            base=remote,
-            need_sync=False,
-            updated=remote.updated,
-            last_processed_message=remote.last_processed_message,
-            workspaces=remote.workspaces,
-            speculative=False,
-        )
-
-    def to_remote(self, author: DeviceID, timestamp: DateTime) -> RemoteUserManifest:
-        return RemoteUserManifest(
-            author=author,
-            timestamp=timestamp,
-            id=self.id,
-            version=self.base_version + 1,
-            created=self.created,
-            updated=self.updated,
-            last_processed_message=self.last_processed_message,
-            workspaces=self.workspaces,
-        )
 
 
-_PyLocalUserManifest = LocalUserManifest
-if not TYPE_CHECKING:
-    try:
-        from libparsec.types import LocalUserManifest as _RsLocalUserManifest
-    except:
-        pass
-    else:
-        LocalUserManifest = _RsLocalUserManifest
+def from_remote(
```

Comment:
should call them `manifest_from_remote`/``manifest_from_remote_with_local_context`

Reply:
done